core/sdk-core/src/test/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncSigningStageTest.java

```diff
@@ -177,8 +234,13 @@ public void execute_selectedAuthScheme_asyncRequestBody_doesSraSignAsync() throw
         AsyncSignRequest<? extends Identity> signRequest = asyncSignRequestCaptor.getValue();
         assertThat(signRequest.identity()).isSameAs(identity);
         assertThat(signRequest.request()).isSameAs(request);
-        assertThat(signRequest.property(SIGNER_PROPERTY)).isEqualTo("value");
         assertThat(signRequest.payload().get()).isSameAs(asyncPayload);
+        assertThat(signRequest.property(SIGNER_PROPERTY)).isEqualTo("value");
+
+        // Assert that the time offset set was zero
+        assertThat(signRequest.property(SharedSignerProperty.SIGNING_CLOCK)).isNotNull();
```

Comment:
I think rather not set the property if offset is 0

Reply:
That will introduce a branch that is not needed the way it's written now, so I think is better just to have a single code path. Can you please help me understand why you think is better not to set it?

Reply:
I thought it is better since the test more fragile and can fail sometimes if for some reason Instant.now is actually beyond the `within` value?

Reply:
The other test is still fragile like that. I think we can have a package-private constructor that takes in a Clock (instead of Clock.systemUTC), so that unit test can use a mock Clock and test exactly, instead of `isCloseTo`.

Reply:
That would address my main concern - test fragility. We could still chose to not set or set the clock property if offset is `0`. 
Outside of that, I was thinking that if there is no need for offset, the SigningStage can let the HttpSigner decide whether it needs a clock at all or not. But I don't feel strongly about that, if the test fragility is addressed.

Reply:
I'd rather tackle the problem when and if arises, I'm know that what you describing is possible but I don't think is likely, for that to happen it would mean that the machine running the tests will need to park the thread for more than 100ms (~ 300 Million instructions). Were the previous tests brittle like actually failing every so often?